src/node/index.ts

```diff
@@ -101,6 +108,53 @@ export default function PluginInspect(options: Options = {}): Plugin {
   const idMap: ResolveIdMap = {}
   const idMapSSR: ResolveIdMap = {}
 
+  // a hack for wrapping connect server stack
+  // see https://github.com/senchalabs/connect/blob/0a71c6b139b4c0b7d34c0f3fca32490595ecfd60/index.js#L50-L55
+  function collectMiddlewarePerf(middlewares: Connect.Server['stack']) {
+    let firstMiddlewareIndex = -1
+    return middlewares.map((middleware, index) => {
+      const { handle, ...rest } = middleware
+      if (typeof handle !== 'function' || !handle.name)
+        return middleware
+      
+      return {
+        handle: async (...middlewareArgs: any[]) => {
+          let req: any, res: any, next: any, err: any
+          if (middlewareArgs.length === 4)
+            [err, req, res, next] = middlewareArgs
+
+          else
+            [req, res, next] = middlewareArgs
+
+          const start = Date.now()
+          const url = req.url?.replace(timestampRE, '').replace(trailingSeparatorRE, '')
+          serverPerf.middleware![url] ??= []
+
+          if (firstMiddlewareIndex < 0)
+            firstMiddlewareIndex = index
+
+          // clear middleware timing
+          if (index === firstMiddlewareIndex)
+            serverPerf.middleware![url] = []
+
+          // @ts-expect-error handle needs 3 or 4 arguments
+          await (middlewareArgs.length === 4 ? handle(err, req, res, next) : handle(req, res, next))
+
+          const total = Date.now() - start
+          const metrics = serverPerf.middleware![url]
+
+          // middleware selfTime = totalTime - next.totalTime
```

Comment:
Should read `prev` here, no?
```suggestion
          // middleware selfTime = totalTime - prev.totalTime
```

Reply:
No, it's `next()` totalTime, thinking about the onion model.